packages/beacon-node/src/network/libp2p/index.ts

```diff
@@ -81,7 +81,12 @@ export async function createNodeJsLibp2p(
         backlog: 5,
         closeServerOnMaxConnections: {
           closeAbove: networkOpts.maxPeers ?? Infinity,
-          listenBelow: networkOpts.maxPeers ?? Infinity,
+          // To stop and start listening based on a fixed number of connections
+          // would make it brittle to the order in which connections are established.
+          // Instead, we stop listening when the number of connections is below the target peers
+          // given that target peers and max peers have a reasonable gap between them and
+          // target peers are always less than the max peers
+          listenBelow: networkOpts.targetPeers ?? -Infinity,
```

Comment:
Does this value need to default to -Infinity or 0?

Reply:
I believe both can work as connection count never gonna be below zero. But with the consistent approach of positive Infinity, I suggest to use the negative Infinity.

Reply:
-Infinity is dangerous because it means that once closeAbove is hit, the listener will never turn back on and we won't ever get inbound connections.